compilation error on Fedora 24

Compiling from latest git.

<pre>
[hunterz@dusan rigs-of-rods]$ make -j4
[  0%] Generate files containing version information
true
true
[  0%] Built target generate_version
[  2%] Built target version_info
[  2%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/Configurator.cpp.o
[ 96%] Built target RoR
In file included from /usr/include/gtk-2.0/gdk/gdkscreen.h:32:0,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:31,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from /home/hunterz/rigs-of-rods/source/configurator/wxutils.h:7,
                 from /home/hunterz/rigs-of-rods/source/configurator/Configurator.cpp:76:
/usr/include/gtk-2.0/gdk/gdktypes.h:114:39: error: conflicting declaration 'typedef struct _GdkDrawable GdkWindow'
 typedef struct _GdkDrawable           GdkWindow;
                                       ^~~~~~~~~
In file included from /usr/include/wx-3.0/wx/choice.h:18:0,
                 from /home/hunterz/rigs-of-rods/source/configurator/wxValueChoice.h:28,
                 from /home/hunterz/rigs-of-rods/source/configurator/Configurator.cpp:28:
/usr/include/wx-3.0/wx/defs.h:3412:31: note: previous declaration as 'typedef struct _GdkWindow GdkWindow'
     typedef struct _GdkWindow GdkWindow;
                               ^~~~~~~~~
In file included from /home/hunterz/rigs-of-rods/source/configurator/wxutils.h:8:0,
                 from /home/hunterz/rigs-of-rods/source/configurator/Configurator.cpp:76:
/home/hunterz/rigs-of-rods/source/configurator/wxutils.h: In function 'size_t getOISHandle(wxWindow*)':
/usr/include/gtk-2.0/gdk/gdkx.h:113:69: error: cannot convert 'GdkWindow* {aka _GdkWindow*}' to 'GdkDrawable* {aka _GdkDrawable*}' for argument '1' to 'XID gdk_x11_drawable_get_xid(GdkDrawable*)'
 #define GDK_WINDOW_XID(win)           (gdk_x11_drawable_get_xid (win))
                                                                     ^
/home/hunterz/rigs-of-rods/source/configurator/wxutils.h:54:17: note: in expansion of macro 'GDK_WINDOW_XID'
  hWnd = (size_t)GDK_WINDOW_XID(gtk_widget_get_window(window->GetHandle()));
                 ^
In file included from /usr/include/wx-3.0/wx/choice.h:18:0,
                 from /home/hunterz/rigs-of-rods/source/configurator/wxValueChoice.h:28,
                 from /home/hunterz/rigs-of-rods/source/configurator/Configurator.cpp:28:
/usr/include/wx-3.0/wx/defs.h:3412:20: note: class type 'GdkWindow {aka _GdkWindow}' is incomplete
     typedef struct _GdkWindow GdkWindow;
                    ^~~~~~~~~~
source/configurator/CMakeFiles/RoRConfig.dir/build.make:62: recipe for target 'source/configurator/CMakeFiles/RoRConfig.dir/Configurator.cpp.o' failed
make[2]: *** [source/configurator/CMakeFiles/RoRConfig.dir/Configurator.cpp.o] Error 1
CMakeFiles/Makefile2:252: recipe for target 'source/configurator/CMakeFiles/RoRConfig.dir/all' failed
make[1]: *** [source/configurator/CMakeFiles/RoRConfig.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2

</pre>

--- a/source/main/MainMenu.cpp
+++ b/source/main/MainMenu.cpp
@@ -139,6 +139,7 @@ void MainMenu::EnterMainMenuLoop()
 
         RoR::App::GetOgreSubsystem()->GetOgreRoot()->renderOneFrame();
 
+#ifdef USE_SOCKETW
         if ((App::GetActiveMpState() == App::MP_STATE_CONNECTED) && RoR::Networking::CheckError())
         {
             Ogre::String title = Ogre::UTFString(_L("Network fatal error: ")).asUTF8();
@@ -149,6 +150,7 @@ void MainMenu::EnterMainMenuLoop()
             RoR::App::GetGuiManager()->GetMainSelector()->Hide();
             RoR::App::GetGuiManager()->GetMainSelector()->Show(LT_Terrain);
         }
+#endif
 
         if (!rw->isActive() && rw->isVisible())
             rw->update(); // update even when in background !

--- a/source/main/gameplay/Character.cpp
+++ b/source/main/gameplay/Character.cpp
@@ -592,6 +592,7 @@ void Character::move(Vector3 offset)
 // Helper function
 void Character::ReportError(const char* detail)
 {
+#ifdef USE_SOCKETW
     Ogre::UTFString username;
     RoRnet::UserInfo info;
     if (!RoR::Networking::GetUserInfo(m_source_id, info))
@@ -605,6 +606,7 @@ void Character::ReportError(const char* detail)
         username.asUTF8_c_str(), m_source_id, m_stream_id);
 
     LOGSTREAM << msg_buf << detail;
+#endif
 }
 
 void Character::sendStreamSetup()
@@ -649,6 +651,7 @@ void Character::sendStreamData()
 
 void Character::receiveStreamData(unsigned int& type, int& source, unsigned int& streamid, char* buffer)
 {
+#ifdef USE_SOCKETW
     if (type == RoRnet::MSG2_STREAM_DATA && m_source_id == source && m_stream_id == streamid)
     {
         auto* msg = reinterpret_cast<Networking::CharacterMsgGeneric*>(buffer);
@@ -690,6 +693,7 @@ void Character::receiveStreamData(unsigned int& type, int& source, unsigned int&
             this->ReportError(err_buf);
         }
     }
+#endif
 }
 
 void Character::updateNetLabelSize()

--- a/source/main/gameplay/OutProtocol.cpp
+++ b/source/main/gameplay/OutProtocol.cpp
@@ -54,18 +54,20 @@ OutProtocol::~OutProtocol(void)
 {
     if (sockfd != 0)
     {
-#if _WIN32
+#ifdef USE_SOCKETW
+#   if _WIN32
         closesocket(sockfd);
-#else
-		close( sockfd );
-#endif
+#   else
+        close( sockfd );
+#   endif
+#endif // USE_SOCKETW
         sockfd = 0;
     }
 }
 
 void OutProtocol::startup()
 {
-#ifdef _WIN32
+#if defined(_WIN32) && defined(USE_SOCKETW)
     SWBaseSocket::SWBaseError error;
 
     // startup winsock
@@ -110,7 +112,7 @@ void OutProtocol::startup()
 
 bool OutProtocol::update(float dt)
 {
-#ifdef _WIN32
+#if defined(_WIN32) && defined(USE_SOCKETW)
     if (!working)
     {
         return false;

--- a/source/main/gameplay/RoRFrameListener.cpp
+++ b/source/main/gameplay/RoRFrameListener.cpp
@@ -2506,7 +2506,7 @@ void RoRFrameListener::EnterGameplayLoop()
         }
 
         RoR::App::GetOgreSubsystem()->GetOgreRoot()->renderOneFrame();
-
+#ifdef USE_SOCKETW
         if ((App::GetActiveMpState() == App::MP_STATE_CONNECTED) && RoR::Networking::CheckError())
         {
             Ogre::String title = Ogre::UTFString(_L("Network fatal error: ")).asUTF8();
@@ -2514,6 +2514,7 @@ void RoRFrameListener::EnterGameplayLoop()
             App::GetGuiManager()->ShowMessageBox(title, msg, true, "OK", true, false, "");
             App::SetPendingAppState(App::APP_STATE_MAIN_MENU);
         }
+#endif
 
         if (!rw->isActive() && rw->isVisible())
             rw->update(); // update even when in background !

--- a/source/main/gui/panels/GUI_MultiplayerClientList.cpp
+++ b/source/main/gui/panels/GUI_MultiplayerClientList.cpp
@@ -23,7 +23,6 @@
 /// @date   7th of September 2009
 
 #ifdef USE_MYGUI
-#ifdef USE_SOCKETW
 
 #include "GUI_MultiplayerClientList.h"
 
@@ -298,6 +297,7 @@ void MpClientList::updateSlot(player_row_t* row, RoRnet::UserInfo c, bool self)
 
 void MpClientList::update()
 {
+#ifdef USE_SOCKETW
     int slotid = 0;
 
     MyGUI::IntSize gui_area = MyGUI::RenderManager::getInstance().getViewSize();
@@ -339,6 +339,7 @@ void MpClientList::update()
 
     int height = lineheight * (slotid + 1);
     mpPanel->setSize(sidebarWidth, height);
+#endif
 }
 
 void MpClientList::clickUserGoIcon(MyGUI::WidgetPtr sender)
@@ -384,5 +385,4 @@ bool MpClientList::IsVisible()
     return mpPanel->getVisible();
 }
 
-#endif // USE_SOCKETW
 #endif // USE_MYGUI

--- a/source/main/gui/panels/GUI_MultiplayerClientList.h
+++ b/source/main/gui/panels/GUI_MultiplayerClientList.h
@@ -23,7 +23,6 @@
 /// @date   18th of July 2010
 
 #ifdef USE_MYGUI
-#ifdef USE_SOCKETW
 
 #pragma once
 
@@ -89,7 +88,6 @@ protected:
     static const int sidebarWidth = 250;
 };
 
-#endif // USE_SOCKETW
 #endif // USE_MYGUI
 
 } // namespace GUI

--- a/source/main/main.cpp
+++ b/source/main/main.cpp
@@ -299,11 +299,13 @@ int main(int argc, char *argv[])
 
                 if (previous_application_state == App::APP_STATE_SIMULATION)
                 {
+#ifdef USE_SOCKETW
                     if (App::GetActiveMpState() == App::MP_STATE_CONNECTED)
                     {
                         RoR::Networking::Disconnect();
                         App::GetGuiManager()->SetVisible_MpClientList(false);
                     }
+#endif
                     gEnv->cameraManager->OnReturnToMainMenu();
                     /* Hide top menu */
                     App::GetGuiManager()->SetVisible_TopMenubar(false);

